### PR TITLE
101 telegraf influx db grafana

### PR DESCRIPTION
**Pull request by lavanyasubbarayalu**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by lavanyasubbarayalu** [comment]
@bmoore-msft - can you please review and let me know if this can be merged or needs any additional details.

File: 101-Telegraf-InfluxDB-Grafana/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    "storageAccountName": "[concat(variables('resourcePrefix'), uniquestring(resourceGroup().id))]",
+    "OSDiskName": "[concat(variables('resourcePrefix'), 'OSDisk')]",
+    "nicName": "[concat(variables('resourcePrefix'), 'VMNic')]",
+    "subnetName": "[concat(variables('resourcePrefix'), 'Subnet')]",
+    "publicIPAddressName": "[concat(variables('resourcePrefix'), 'PublicIP')]",
+    "vmStorageAccountContainerName": "vhds",    
+    "vmName": "[concat(variables('resourcePrefix'), 'VM')]",
+    "vmExtensionName": "[concat(variables('resourcePrefix'), 'Init')]",
+    "virtualNetworkName": "[concat(variables('resourcePrefix'), 'VNET')]",
+    "_artifactsLocation": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-Telegraf-InfluxDB-Grafana/",
```
Comment (bmoore-msft) [inline 101-Telegraf-InfluxDB-Grafana/azuredeploy.json]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc) for how to handle artifacts

File: 101-Telegraf-InfluxDB-Grafana/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    "_extensionScript": "tigscript.sh",
+    "_artifactsLocationSasToken": "",
+	"_configfilelocation": "[concat(variables('_artifactsLocation'), 'scripts/Configfiles.zip')]"
```
Comment (bmoore-msft) [inline 101-Telegraf-InfluxDB-Grafana/azuredeploy.json]: format the json, e.g. SHIFT+ALR+F in VS Code

File: 101-Telegraf-InfluxDB-Grafana/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('storageAccountName')]",
+      "apiVersion": "2016-01-01",
```
Comment (bmoore-msft) [inline 101-Telegraf-InfluxDB-Grafana/azuredeploy.json]: update apiVersion

File: 101-Telegraf-InfluxDB-Grafana/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        "ipConfigurations": [
+          {
+            "name": "ipconfig1",
+            "properties": {
+              "privateIPAllocationMethod": "Dynamic",
+              "publicIPAddress": {
+                "id": "[resourceId('Microsoft.Network/publicIPAddresses', variables('publicIPAddressName'))]"
+              },
+              "subnet": {
+                "id": "[concat(resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName')),'/subnets/',variables('subnetName'))]"
```
Comment (bmoore-msft) [inline 101-Telegraf-InfluxDB-Grafana/azuredeploy.json]: remove concat, just use resourceId

File: 101-Telegraf-InfluxDB-Grafana/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+              }
+            }
+          }
+        ]
+      }
+    },
+
+    {
+      "apiVersion": "2015-06-15",
```
Comment (bmoore-msft) [inline 101-Telegraf-InfluxDB-Grafana/azuredeploy.json]: update apiVersion

File: 101-Telegraf-InfluxDB-Grafana/azuredeploy.parameters.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "adminUsername": {
+      "value": "GEN-USERNAME"
+    },
+    "sshPublicKey": {
+      "value": "GEN-SSH-PUB-KEY"
+    },
+    "tigDnsPrefix": {
+      "value": "gen-unique-3"
```
Comment (bmoore-msft) [inline 101-Telegraf-InfluxDB-Grafana/azuredeploy.parameters.json]: remove the -3 here, 3 chars isn't very unique and there doesn't seem to be any reason to restrict it

**Comment by lavanyasubbarayalu** [comment]
@bmoore-msft - updated the files with review comments fix.

File: 101-Telegraf-InfluxDB-Grafana/azuredeploy.json [hunk 7]
```diff
@@ -1,210 +1,197 @@
...
+        "storageAccountName": "[concat(variables('resourcePrefix'), uniquestring(resourceGroup().id))]",
+        "OSDiskName": "[concat(variables('resourcePrefix'), 'OSDisk')]",
+        "nicName": "[concat(variables('resourcePrefix'), 'VMNic')]",
+        "subnetName": "[concat(variables('resourcePrefix'), 'Subnet')]",
+        "publicIPAddressName": "[concat(variables('resourcePrefix'), 'PublicIP')]",
+        "vmStorageAccountContainerName": "vhds",
+        "vmName": "[concat(variables('resourcePrefix'), 'VM')]",
+        "vmExtensionName": "[concat(variables('resourcePrefix'), 'Init')]",
+        "virtualNetworkName": "[concat(variables('resourcePrefix'), 'VNET')]",
+        "_artifactsLocation": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-Telegraf-InfluxDB-Grafana/",
```
Comment (bmoore-msft) [inline 101-Telegraf-InfluxDB-Grafana/azuredeploy.json]: these must be parameters as per the BP guide

File: 101-Telegraf-InfluxDB-Grafana/azuredeploy.json [hunk 8]
```diff
@@ -1,210 +1,197 @@
...
+                    }
+                }
             },
-            "protectedSettings": {
-              "commandToExecute": "[concat('sh tigscript.sh',' ',variables('_configfilelocation'))]"
-            }
+            "resources": [{
+                "type": "extensions",
+                "name": "[variables('vmExtensionName')]",
+                "apiVersion": "2015-06-15",
```
Comment (bmoore-msft) [inline 101-Telegraf-InfluxDB-Grafana/azuredeploy.json]: update apiVersion

**Comment by lavanyasubbarayalu** [comment]
@bmoore-msft - updated with review comments fix.

**Comment by lavanyasubbarayalu** [comment]
@bmoore-msft - i have fixed the requested changes, can you please check the same for merging. thanks.

File: 101-Telegraf-InfluxDB-Grafana/azuredeploy.json [hunk 9]
```diff
@@ -1,210 +1,209 @@
...
+                    "[resourceId('Microsoft.Compute/virtualMachines', variables('vmName'))]"
+                ],
+                "properties": {
+                    "publisher": "Microsoft.Azure.Extensions",
+                    "type": "CustomScript",
+                    "typeHandlerVersion": "2.0",
+                    "autoUpgradeMinorVersion": true,
+                    "settings": {
+                        "fileUris": [
+                            "[concat(parameters('_artifactsLocation'),'scripts/',variables('_extensionScript'))]"
```
Comment (bmoore-msft) [inline 101-Telegraf-InfluxDB-Grafana/azuredeploy.json]: need to include the sasToken

**Comment by lavanyasubbarayalu** [comment]
@bmoore-msft  - i have fixed the requested changes, thanks for your review comments.

**Comment by lavanyasubbarayalu** [comment]
@bmoore-msft - can you please check for merging.Thanks lavanya

**Pull request merged**